From 8499d651d9c3472763b43370e0d1889943708239 Mon Sep 17 00:00:00 2001
From: Andrei Stefanescu <andrei.stefanescu@nxp.com>
Date: Wed, 1 Feb 2023 13:32:45 +0200
Subject: [PATCH 33/54] s32: scmi: add support for pinctrl vendor extension
 protocol

Add support for the pinctrl vendor extension that forwards pinctrl
requests to the SCP firmware via SCMI.

Issue: ALB-9426

Upstream-Status: Pending 

Signed-off-by: Andrei Stefanescu <andrei.stefanescu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 plat/nxp/s32/include/s32_scp_scmi.h | 5 +++--
 plat/nxp/s32/s32_scp_scmi.c         | 3 ++-
 plat/nxp/s32/s32_svc.c              | 3 ++-
 3 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/plat/nxp/s32/include/s32_scp_scmi.h b/plat/nxp/s32/include/s32_scp_scmi.h
index 06887a35b..1a1a3654f 100644
--- a/plat/nxp/s32/include/s32_scp_scmi.h
+++ b/plat/nxp/s32/include/s32_scp_scmi.h
@@ -1,12 +1,13 @@
 /*
- * Copyright 2022 NXP
+ * Copyright 2022-2023 NXP
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
 #ifndef S32_SCP_SCMI_H
 #define S32_SCP_SCMI_H
 
-#define SCMI_PROTOCOL_ID_GPIO	(0x81u)
+#define SCMI_PROTOCOL_ID_PINCTRL	(0x80u)
+#define SCMI_PROTOCOL_ID_GPIO		(0x81u)
 
 typedef int (*scmi_msg_callback_t)(void *payload);
 
diff --git a/plat/nxp/s32/s32_scp_scmi.c b/plat/nxp/s32/s32_scp_scmi.c
index f823c5dd0..341048548 100644
--- a/plat/nxp/s32/s32_scp_scmi.c
+++ b/plat/nxp/s32/s32_scp_scmi.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2022 NXP
+ * Copyright 2022-2023 NXP
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -380,6 +380,7 @@ static bool is_proto_allowed(mailbox_mem_t *mbx_mem)
 	case SCMI_PROTOCOL_ID_PERF:
 	case SCMI_PROTOCOL_ID_CLOCK:
 	case SCMI_PROTOCOL_ID_RESET_DOMAIN:
+	case SCMI_PROTOCOL_ID_PINCTRL:
 	case SCMI_PROTOCOL_ID_GPIO:
 		return true;
 	}
diff --git a/plat/nxp/s32/s32_svc.c b/plat/nxp/s32/s32_svc.c
index 7c35e508c..28931bcf8 100644
--- a/plat/nxp/s32/s32_svc.c
+++ b/plat/nxp/s32/s32_svc.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2020-2022 NXP
+ * Copyright 2020-2023 NXP
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -41,6 +41,7 @@ static const uint8_t s32_protocols[] = {
 	SCMI_PROTOCOL_ID_PERF,
 	SCMI_PROTOCOL_ID_CLOCK,
 	SCMI_PROTOCOL_ID_RESET_DOMAIN,
+	SCMI_PROTOCOL_ID_PINCTRL,
 	SCMI_PROTOCOL_ID_GPIO,
 	0U,
 };
-- 
2.25.1

